### ISSUE TITLE
CSRF protection

### ISSUE THREAD

**Issue by forgetaboutit** [muhbaasu/pfennig-server#20]
Issue CSRF tokens in server-generated forms to protect against [cross-site request forgery](https://en.wikipedia.org/wiki/Cross-site_request_forgery)

**Comment by marcelbuesing** [comment]
Good point! Looks like the "Synchronizer token pattern" should suffice for now. I guess REST requests should not be given a JWT via a cookie to avoid establishing a trust relationship over multiple requests. They will have to send it in the HTTP-Header for each request.

**Comment by marcelbuesing** [comment]
For Form Url encoded requests use the Synchronizer token pattern, otherwise use "Cookie-to-Header" Token, which means read a specific http header from the request.

**Comment by forgetaboutit** [comment]
Here's a [post](http://security.stackexchange.com/questions/62769/must-login-and-logout-action-have-csrf-protection) describing why logins should be CSRF-protected as well.

**Comment by marcelbuesing** [comment]
Stateless CSRF [post](http://appsandsecurity.blogspot.de/2012/01/stateless-csrf-protection.html).
